api/sign_transaction.go

```diff
@@ -27,21 +26,18 @@ func signTxFromArgs(args TransactionArgs) ([]byte, error) {
 		data = *args.Input
 	}
 
-	// provide a high enough gas for the tx to be able to execute
+	// provide a high enough gas for the tx to be able to execute,
+	// capped by the gas set in transaction args.
 	gasLimit := defaultGasLimit
 	if args.Gas != nil {
 		gasLimit = uint64(*args.Gas)
 	}
-	value := big.NewInt(0)
-	if args.Value != nil {
-		value = args.Value.ToInt()
-	}
 
 	tx := types.NewTx(
 		&types.LegacyTx{
 			Nonce:    0,
 			To:       args.To,
-			Value:    value,
+			Value:    big.NewInt(0),
```

Comment:
Agreed :+1: Most likely I will open an issue in flow-go, since there's no way to achieve this functionality without exposing a relevant method on the `EVM` contract.